library/ui/src/main/java/com/google/android/exoplayer2/ui/TrackSelectionView.java

```diff
@@ -367,6 +380,48 @@ private void onTrackViewClicked(View view) {
     }
   }
 
+  private TrackGroupArray initSortedTrackGroups(TrackGroupArray trackGroups) {
```

Comment:
I don't think this method does quite the right thing. In particular, suppose that there are multiple groups in the `TrackGroupArray`. L394 below rewrites it so there's only a single group. Doesn't this cause all other groups to be lost?

Reply:
> L394 below rewrites it so there's only a single group. Doesn't this cause all other groups to be lost?

Oh I see what you mean, I haven't been able to find a `TrackGroupArray` that has more than one group.
Yes this needs slight modification, currently it only saves the last group, I'll change it so it keeps all the groups and sort them according to the `comparator`

Reply:
> L394 below rewrites it so there's only a single group. Doesn't this cause all other groups to be lost?

Oh I see what you mean, I haven't been able to find a `TrackGroupArray` that has more than one group.
Yes this needs slight modification, currently it only saves the last group, I'll change it so it keeps all the groups and sort them according to the `comparator`